### PR TITLE
Add Cake

### PR DESCRIPTION
**Pull request by tmagicturtle**

### CODE REVIEW

File: data/mods/ssb/abilities.ts [hunk 1]
```diff
@@ -51,6 +51,53 @@ export const BattleAbilities: {[k: string]: ModdedAbilityData} = {
...
+			let typeList = ["Normal", "Fighting", "Flying", "Poison", "Ground", "Rock",
+								"Bug", "Ghost", "Steel", "Fire", "Water", "Grass", "Electric",
+								"Psychic", "Ice", "Dragon", "Dark", "Fairy"];
+			this.prng.shuffle(typeList);
+			let firstType = typeList[0];
+			this.prng.shuffle(typeList);
+			let secondType = typeList[0];
+			let newTypes = [];
+			if (firstType === secondType) {
+				newTypes = [firstType]
```
Comment (ghost) [inline data/mods/ssb/abilities.ts]: oops

File: data/mods/ssb/moves.ts [hunk 2]
```diff
@@ -96,6 +96,47 @@ export const BattleMovedex: {[k: string]: ModdedMoveData} = {
...
+		name: "Kevin",
+		pp: 10,
+		flags: {protect: 1, mirror: 1},
+		onTryMove() {
+			this.attrLastMove('[still]');
+		},
+		onPrepareHit(target, source) {
+			this.add('-anim', source, 'Brave Bird', target);
+		},
+		onModifyMove(move, pokemon, target) {
```
Comment (ghost) [inline data/mods/ssb/moves.ts]: we have onModifyType

Comment (ghost) [reply data/mods/ssb/moves.ts]: also, I would rather make the first type be the type of the move, and the second type be in the effectiveness.  You dont need to bend over backwards to make the effectiveness for 2 types.

File: data/mods/ssb/statuses.ts [hunk 3]
```diff
@@ -62,6 +62,21 @@ export const BattleStatuses: {[k: string]: ModdedPureEffectData} = {
...
+			this.add(`c|${getName('Cake')}|AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA`);
+			if (source.species.id !== 'dunsparce' || source.illusion) return;
+			// Cake innate
+			this.boost({atk: 1, def: 1, spa: 1, spd: 1, spe: 1}, source, source);
+		},
+		onSwitchOut() {
+			this.add(`c|${getName('Cake')}|${pokemon.side.name} is a nerd`);
+		},
+		onFaint() {
+			this.add(`c|${getName('Cake')}|Chowder was a good show}`);
```
Comment (ghost) [inline data/mods/ssb/statuses.ts]: missed a curly

File: data/mods/ssb/statuses.ts [hunk 4]
```diff
@@ -62,6 +62,21 @@ export const BattleStatuses: {[k: string]: ModdedPureEffectData} = {
 			this.add(`c|${getName('Alpha')}|caio`);
 		},
 	},
+	cake: {
+		noCopy: true,
+		onStart(target, source) {
+			this.add(`c|${getName('Cake')}|AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA`);
+			if (source.species.id !== 'dunsparce' || source.illusion) return;
```
Comment (ghost) [inline data/mods/ssb/statuses.ts]:
```suggestion
			if (source.illusion) return;
```

Comment (ghost) [reply data/mods/ssb/statuses.ts]: Only modify species if you are using megas.

File: data/mods/ssb/abilities.ts (hunk 1 above)
Comment (tmagicturtle) [reply data/mods/ssb/abilities.ts]: What's the issue here? It's so that it doesn't show as Electric/Electric.

Comment (ghost) [reply data/mods/ssb/abilities.ts]:
```suggestion
				newTypes = [firstType];
```

**Pull request closed without merging**